flows/interfaces.go

```diff
@@ -163,12 +163,17 @@ type Context interface {
 }
 
 type Event interface {
-	CreatedOn() *time.Time
+	CreatedOn() time.Time
 	SetCreatedOn(time.Time)
 
 	Step() StepUUID
 	SetStep(StepUUID)
 
+	FromCaller() bool
+	SetFromCaller(bool)
```

Comment:
Wonder if this, step and create_on are other signs that maybe we need something wrapping an Event that is really the StepEvent or the like.. would that make sense?

So while we currently have stuff like:

```json
{ "events": "events": [
                  {
                    "created_on": "2000-01-01T00:00:00.000000000-00:00",
                    "groups": [
                      {
                        "name": "Survey Audience",
                        "uuid": "2aad21f6-30b7-42c5-bd7f-1b720c154817"
                      }
                    ],
                    "step_uuid": "",
                    "type": "add_to_group"
                  },
                  {
                    "created_on": "2000-01-01T00:00:00.000000000-00:00",
                    "field_name": "Activation Token",
                    "field_uuid": "f06c5b73-eb0d-4417-b7e0-4f650ed30dc8",
                    "step_uuid": "",
                    "type": "save_contact_field",
                    "value": "XXX-YYY-ZZZ"
                  }
]
}
```

Maybe it is more right to have something like:

```json
{ "events": "events": [
                  {
                    "created_on": "2000-01-01T00:00:00.000000000-00:00",
                    "step_uuid": "",
                    "event": {
                    "groups": [
                      {
                        "name": "Survey Audience",
                        "uuid": "2aad21f6-30b7-42c5-bd7f-1b720c154817"
                      }
                    ],
                    "type": "add_to_group"
                    }
                  },
                  {
                    "created_on": "2000-01-01T00:00:00.000000000-00:00",
                    "step_uuid": "",
                    "event": {
                    "field_name": "Activation Token",
                    "field_uuid": "f06c5b73-eb0d-4417-b7e0-4f650ed30dc8",
                    
                    "type": "save_contact_field",
                    "value": "XXX-YYY-ZZZ"
                  }
   }
]
}
```

It definitely is less than ideal that we have `step_uuid` on event for example, when this is really only an attribute that is there because of the passed back `events` list, not really an attribute of event at all. Basically maybe we should be putting events in containers that contain timestamps / step uuids instead of making them all deal with that. (and then having the confusion as to whether that is a required attribute by the caller)

Any ideas there?